docs/docs/casper/overview.md

```diff
@@ -0,0 +1,145 @@
+---
+title: Overview
+
+slug: /casper/project_structure
+---
+
+This section provides an overview of the structural organization of the Invariant Protocol smart contract project on Casper. The project is meticulously structured to enhance readability, maintainability, and efficiency. The architecture is designed to consolidate data within a single contract, minimizing fees and simplifying interactions.
+
+## Contract Architecture
+
+To optimize gas usage, we centralize data and entrypoints in a singular contract, reducing expenses associated with pool and position creation. This streamlined approach not only cuts costs but also simplifies processes, enhancing accessibility. By concentrating state changes and entrypoints within this central contract, we eliminate the intricacies of managing external contracts, while smart mapping intelligently conserves storage resources and bolsters system efficiency.
+
+## Project Structure
+
+The following presents the project's overall structure, supplying insights into the logical segmentation into modules:
+
+```
+📦protocol-cspr
+ ┣ 📂contracts
+ ┃ ┣ 📂collections
+ ┃ ┣ 📂logic
+ ┃ ┣ 📂storage
+ ┃ ┗ 📜entrypoints.rs
```

Comment:
it's worth it to include: `contracts/events.rs`, `contracts/errors.rs` inside structure and below in descriptions